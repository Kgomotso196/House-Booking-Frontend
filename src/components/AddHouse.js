import React, { useState } from 'react';

const AddHouse = () => {
  const [formData, setFormData] = useState({
    house_name: '',
    house_image: '',
  });

  const handleSubmit = async (event) => {
    event.preventDefault();
    try {
      const response = await fetch('http://3001/api/v1/houses', {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json',
        },
        body: JSON.stringify(formData),
      });
      if (response.ok) {
        // Redirecting user here
      } else {
        // Handle error here
      }
    } catch (error) {
      // Error handling here
    }
  };

  const handleChange = (event) => {
    const { name, value } = event.target;
    setFormData({ ...formData, [name]: value });
  };

  return (
    <div>
      <h1>Add House</h1>
      <form onSubmit={handleSubmit}>
        <input type="text" name="house_name" value={formData.house_name} onChange={handleChange} placeholder="House Name" />
        {/* Details coming soon */}
        <button type="submit">Add House</button>
      </form>
    </div>
  );
};

export default AddHouse;
