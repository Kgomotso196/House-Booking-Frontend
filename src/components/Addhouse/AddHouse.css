.add-house {
  display: flex;
  flex-direction: column;
  width: 80%;
  margin: auto;
  margin-left: 300px;
  justify-items: center;
  align-items: center;
}

.add-house-form {
  display: flex;
  flex-direction: column;
  width: 90%;
}

.btn {
  background-color: #97bf0f;
  border: none;
}

.btn:hover {
  background-color: green;
}

.form-label {
  color: #000;
  margin-bottom: 5px;
}

.error-message {
  color: red;
  font-size: 14px;
}

.input-error {
  border: 2px solid red;
  box-shadow: 0 0 5px red;
}

@media (max-width: 600px) {
  .add-house {
    width: 100%;
    margin-left: 5%;
  }

  .add-house-form {
    width: 100%;
  }
}
