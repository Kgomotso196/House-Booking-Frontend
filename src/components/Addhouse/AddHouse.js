import React, { useState } from 'react';
import { useDispatch } from 'react-redux';
import { addHouse } from '../../redux/slices/PostHousesSlice';

const AddHouse = () => {
  const dispatch = useDispatch();
  const [houseData, setHouseData] = useState({
    name: '',
    image: '',
    location: '',
    description: '',
  });

  const handleChange = (e) => {
    const { name, value } = e.target;
    setHouseData({ ...houseData, [name]: value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();

    // Check if any of the form fields are empty
    if (
      !houseData.name
      || !houseData.image
      || !houseData.location
      || !houseData.description
    ) {
      // Handle the case where not all values are present (e.g., show an error message)
      console.log('Please fill in all fields.');
    } else {
      // All values are present, you can proceed with form submission
      // Add your logic to handle the form submission here
      console.log('House Data', houseData);
      try {
        dispatch(addHouse(houseData));
      } catch (error) {
        console.error('Error dispatching the addHouse action:', error);
      }
    }
  };

  return (
    <div className="add-house">
      <h1 className="text-primary">Add New House</h1>
      <form className="add-house-form" onSubmit={handleSubmit}>
        <label htmlFor="houseName" className="form-label text-primary">
          House Name
          <input
            type="text"
            name="name"
            value={houseData.name}
            onChange={handleChange}
            id="houseName"
            className="form-control"
            placeholder="Enter House Name"
          />
        </label>
        <label htmlFor="houseimage" className="form-label text-primary">
          House Image
          <input
            type="text"
            name="image"
            value={houseData.image}
            onChange={handleChange}
            id="houseimage"
            className="form-control"
            placeholder="Enter House image"
          />
        </label>
        <label htmlFor="houseLocation" className="form-label text-primary">
          House Location
          <input
            type="text"
            name="location"
            value={houseData.location}
            onChange={handleChange}
            id="houseLocation"
            className="form-control"
            placeholder="Enter House Location"
          />
        </label>
        <label htmlFor="houseDescription" className="form-label text-primary">
          Description
          <input
            type="text"
            name="description"
            value={houseData.description}
            onChange={handleChange}
            id="houseDescription"
            className="form-control"
            placeholder="Enter Description"
          />
        </label>
        <div>
          <input type="submit" className="btn btn-primary" value="Add House" />
        </div>
      </form>
    </div>
  );
};

export default AddHouse;
