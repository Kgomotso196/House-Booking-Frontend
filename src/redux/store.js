import { configureStore } from '@reduxjs/toolkit';
import getReservationSlice from './slices/GetReservationSlice';
import getHosesSlice from './slices/GetHosesSlice';
import postReservationSlice from './slices/PostReservationSlice';
import postHousesSlice from './slices/PostHousesSlice';

const store = configureStore({
  reducer: {
    reservations: getReservationSlice,
    houses: getHosesSlice,
    postReservations: postReservationSlice,
    postHouses: postHousesSlice,
  },
});

export default store;
