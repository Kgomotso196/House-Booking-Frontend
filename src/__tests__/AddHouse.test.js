/* eslint-env jest */
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../redux/store';
import AddHouse from '../components/Addhouse/AddHouse';

describe('Test AddHouse Component', () => {
  test('it renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <AddHouse />
          </Router>
        </Provider>,
      ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
