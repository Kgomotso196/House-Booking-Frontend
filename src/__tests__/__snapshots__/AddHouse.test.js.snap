// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test AddHouse Component it renders correctly 1`] = `
<div
  className="add-house"
>
  <h1>
    ADD A NEW HOUSE
  </h1>
  <form
    className="add-house-form"
    onSubmit={[Function]}
  >
    <label
      className="form-label"
      htmlFor="houseName"
    >
      House Name
      <input
        className="form-control"
        id="houseName"
        name="name"
        onChange={[Function]}
        placeholder="Enter House Name"
        type="text"
        value=""
      />
    </label>
    <label
      className="form-label"
      htmlFor="houseimage"
    >
      House Image
      <input
        className="form-control "
        id="houseimage"
        name="image"
        onChange={[Function]}
        placeholder="Enter House image"
        type="text"
        value=""
      />
    </label>
    <label
      className="form-label"
      htmlFor="houseLocation"
    >
      House Location
      <input
        className="form-control"
        id="houseLocation"
        name="location"
        onChange={[Function]}
        placeholder="Enter House Location"
        type="text"
        value=""
      />
    </label>
    <label
      className="form-label"
      htmlFor="houseDescription"
    >
      Description
      <input
        className="form-control"
        id="houseDescription"
        name="description"
        onChange={[Function]}
        placeholder="Enter Description"
        type="text"
        value=""
      />
    </label>
    <div>
      <input
        className="btn btn-primary"
        type="submit"
        value="ADD HOUSE"
      />
    </div>
  </form>
</div>
`;
